test/functional/services/browser.ts

```diff
@@ -213,70 +189,47 @@ export async function BrowserProvider({ getService }: FtrProviderContext) {
-      // tslint:disable-next-line:variable-name
-      const _convertPointW3C = async (point: any, offset: { x: any; y: any }) => {
-        if (point.location instanceof WebElementWrapper) {
-          const position = await point.location.getPosition();
-          return {
-            x: Math.round(position.x + offset.x),
-            y: Math.round(position.y + offset.y),
-          };
-        } else {
-          return {
-            x: Math.round(point.location.x + offset.x),
-            y: Math.round(point.location.y + offset.y),
-          };
-        }
-      };
-      // tslint:disable-next-line:variable-name
-      const _convertPoint = (point: any) => {
-        return point.location instanceof WebElementWrapper
-          ? point.location._webElement
-          : point.location;
-      };
-
       if (this.isW3CEnabled) {
-        // tslint:disable-next-line:variable-name
-        _from = await _convertPointW3C(from, _fromOffset);
-        // tslint:disable-next-line:variable-name
-        _to = await _convertPointW3C(to, _toOffset);
-        // tslint:disable-next-line:variable-name
-        const _offset = { x: _to.x - _from.x, y: _to.y - _from.y };
+        // The offset should be specified in pixels relative to the center of the element's bounding box
+        const getW3CPoint = (data: any) => {
+          if (!data.offset) {
+            data.offset = {};
+          }
+          return data.location instanceof WebElementWrapper
+            ? { x: data.offset.x || 0, y: data.offset.y || 0, origin: data.location._webElement }
+            : { x: data.location.x, y: data.location.y, origin: 'pointer' };
+        };
 
+        const startPoint = getW3CPoint(from);
+        const endPoint = getW3CPoint(to);
+        await this.getActions()
+          .move({ x: 0, y: 0 })
+          .perform();
         return await this.getActions()
-          .move({ x: _from.x, y: _from.y, origin: 'pointer' })
+          .move(startPoint)
           .press()
-          .move({ x: _offset.x, y: _offset.y, origin: 'pointer' })
+          .move(endPoint)
           .release()
           .perform();
       } else {
-        // until Chromedriver is not supporting W3C Webdriver Actions API
-        // tslint:disable-next-line:variable-name
-        _from = _convertPoint(from);
-        // tslint:disable-next-line:variable-name
-        _to = _convertPoint(to);
-        if (from.location instanceof WebElementWrapper && typeof to.location.x === 'number') {
+        // The offset should be specified in pixels relative to the top-left corner of the element's bounding box
+        const getOffset: any = (offset: { x: number; y: number }) =>
+          offset ? { x: offset.x, y: offset.y } : { x: 0, y: 0 };
+
+        if (from.location instanceof WebElementWrapper === false) {
+          throw new Error('Dragging point should be WebElementWrapper instance');
```

Comment:
To avoid confusing while getting this error. `InvalidArgumentError: invalid argument: at least an element or offset should be set
         │   (Session info: chrome=76.0.3809.132)
         │   (Driver info: chromedriver=76.0.3809.68 (420c9498db8ce8fcd190a954d51297672c1515d5-refs/branch-heads/3809@{#864}),platform=Mac OS X 10.14.6 x86_64)`

In non-w3c start point should be provided as WebElement